### ISSUE TITLE
Unable to download video if the video is in a QRT

### ISSUE THREAD

**Issue by biggestsonicfan** [AlttiRi/twitter-click-and-save#7]
Possibly a result of fixing #4, the media in the QRT of [this tweet](https://twitter.com/kuvosa/status/1562180588307509249) cannot be downloaded as a video. The image preview, however, downloads fine.

**Comment by AlttiRi** [comment]
> Possibly a result of fixing #4
Possibly no, since in this strange case there is no link (an achor element `<a href="...">`) to the embed tweet.
So the UJS just uses the parent tweet ID, so it works like before #4.

**Comment by AlttiRi** [comment]
Okay, the normal video tweet reposts are still downloaded as earlier:

![image](https://user-images.githubusercontent.com/16310547/186579906-5e3841b1-5314-487c-9b2f-fe2b0cbf3f34.png)

---

Your [URL](https://twitter.com/kuvosa/status/1562180588307509249) is strange since it does not contain the URL to the embed tweet in HTML, so I can't detect the target tweet ID.

_Hm, why is this retweet so special?_

However, if I open the embed tweet by the middle mouse click, it's just a normal video:
![image](https://user-images.githubusercontent.com/16310547/186580904-163ceebc-236e-4d4f-a10b-f3118510b1e7.png)


---

https://twitter.com/SpaceX/status/1560707309361172480 is a "broadcast" and it can't be download as a usual video.

---

BTW, gallery-dl and other downloaders also do not support these URLs (and [adv](https://github.com/AlttiRi/twitter-click-and-save/issues/6) too).

---

It seems that I need to use an other API for that rare cases.

**Comment by AlttiRi** [comment]
Okay, I don't think that it's critical (the userscript still downloads the usual videos and images), so I will try to fix your last two issues some time later.

It seem I just need to use `https://twitter.com/i/api/graphql/7i51yEC2i_iurfMKZ8mcNw/TweetDetail` endpoint.
_I'm so lazy do it, but okay._

---

Also it seems "broadcast"s do not have a ".mp4" URL, only HLS chunk stream (".m3u8" URL).
At least in the `https://twitter.com/i/api/1.1/live_video_stream/status/` endpoint that Twitter uses.

**Comment by cglmrfreeman** [comment]
> Okay, I don't think that it's critical

I completely agree, these are definitely non-critical things, but are just things I would like to bring to your attention to address functionality of the script. Especially if you were never made aware of them, as was the case with the adv tweet.

I greatly appreciate your time and effort in this script, as I avidly used to right-click tweets, but this has made my life much easier!

**Comment by cglmrfreeman** [comment]
In perhaps an ironic twist of fate, the ability to save that video is now no longer possible as it has been removed by a copyright takedown. If you have the source information you posted previously saved, you can use that to continue development, else this issue will have to be revisited once a similar tweet is found...

**Comment by cglmrfreeman** [comment]
[New one](https://twitter.com/90sManiax/status/1567895914593001473)

**Comment by cglmrfreeman** [comment]
Found a [new video](https://twitter.com/llluicent/status/1572119799219781633) that won't download as QRT but downloads fine in the original.

**Comment by AlttiRi** [comment]
Should be fixed in GitHub dev version.

No need to even use GraphQL API.

**Comment by cglmrfreeman** [comment]
Inverse just happened with the video as the QRT [here](https://twitter.com/Sin_Salt/status/1586455005506449409).

**Comment by AlttiRi** [comment]
Seems to be fixed.

**Comment by cglmrfreeman** [comment]
> Seems to be fixed in [06a13b2](https://github.com/AlttiRi/twitter-click-and-save/commit/06a13b2f2c6199b0938d81d16c4a8366dd9dc1e8).

I wish every git repo worked as hard and as fast as you :laughing:

**Comment by AlttiRi** [comment]
Fixed again. https://github.com/AlttiRi/twitter-click-and-save/commit/92de3f2c0e24dba63e147c9f3823786d5038d7c2
However, it's seems I need to put into the history both tweet IDs: of the original tweet and of the quoted post.

**Comment by AlttiRi** [comment]
https://github.com/AlttiRi/twitter-click-and-save/commit/78164e00485cf0496e0202c58497384f9c2a7741
